pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -0,0 +1,241 @@
+	if err != nil {
+		return nil, err
+	}
+
+	// temporary workaround until store has implemented read/write shipper interface
+	c.bloomShipperClient = client
+	c.bloomStore = store
+	c.Service = services.NewIdleService(c.starting, c.stopping)
+
+	return c, nil
+}
+
+func (c *Compactor) starting(_ context.Context) error {
+	return nil
+}
+
+func (c *Compactor) stopping(_ error) error {
+	return nil
+}
+
+// TODO Get fpRange owned by the compactor instance
+func NoopGetFingerprintRange() (uint64, uint64) { return 0, 0 }
+
+// TODO List Users from TSDB and add logic to owned user via ring
+func NoopGetUserId() string { return "" }
+
+// TODO get series from objectClient (TSDB) instead of params
+func NoopGetSeries() *v1.Series { return nil }
+
+// TODO Then get chunk data from series
+func NoopGetChunks() []byte { return nil }
+
+// part1: Create a compact method that assumes no block/meta files exists (eg first compaction)
+// part2: Write logic to check first for existing block/meta files and does above.
+func (c *Compactor) compactNewChunks(ctx context.Context, dst string) (err error) {
+	//part1
+	series := NoopGetSeries()
+	data := NoopGetChunks()
+
+	// TODO discuss with Owen is blooms meant to be created here, when should tokenization happen?
+	bloom := v1.Bloom{} // TODO {sbf: sbf} sbf is not public, is it meant to be?
+	// create bloom filters from that.
+	sbf := *boom.NewDefaultScalableBloomFilter(0.01)
+	sbf.Add([]byte(fmt.Sprint(data)))
+
+	// block and seriesList
+	seriesList := []v1.SeriesWithBloom{
+		{
+			Series: series,
+			Bloom:  &bloom,
+		},
+	}
+
+	writer := v1.NewDirectoryBlockWriter(dst)
+
+	builder, err := v1.NewBlockBuilder(
+		v1.BlockOptions{
+			SeriesPageSize: 100,
+			BloomPageSize:  10 << 10,
+		}, writer)
+	if err != nil {
+		return err
+	}
+	// BuildFrom closes itself
+	err = builder.BuildFrom(v1.NewSliceIter[v1.SeriesWithBloom](seriesList))
+	if err != nil {
+		return err
+	}
+
+	// TODO Ask Owen, shall we expose a method to expose these paths on BlockWriter?
+	indexPath := filepath.Join(dst, "series")
+	bloomPath := filepath.Join(dst, "bloom")
+
+	blockRef := bloomshipper.BlockRef{
+		IndexPath: indexPath,
+		BlockPath: bloomPath,
+	}
+
+	blocks := []bloomshipper.Block{
+		{
+			BlockRef: blockRef,
+
+			// TODO point to the data to be read
+			Data: nil,
+		},
+	}
+
+	meta := bloomshipper.Meta{
+		// After successful compaction there should be no tombstones
+		Tombstones: make([]bloomshipper.BlockRef, 0),
+		Blocks:     []bloomshipper.BlockRef{blockRef},
+	}
+
+	c.bloomShipperClient.PutMeta(ctx, meta)
+	c.bloomShipperClient.PutBlocks(ctx, blocks)
+	return nil
+}
+
+func (c *Compactor) runCompact(ctx context.Context) error {
+	//TODO set MaxLookBackPeriod to Max ingester accepts
+	maxLookBackPeriod := c.cfg.MaxLookBackPeriod
+
+	st_fp, end_fp := NoopGetFingerprintRange()
+	tenantId := NoopGetUserId()
+
+	end := time.Now().UTC().UnixMilli()
+	start := end - maxLookBackPeriod.Milliseconds()
```

Comment:
Our timestamps are nanosecond precision:

```suggestion
	end := time.Now().UTC().Nanos()
	start := end - maxLookBackPeriod.Nanos()
```